cpp/examples/symreg/symreg_example.cpp

```diff
@@ -0,0 +1,404 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <algorithm>
+#include <cmath>
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+#include <vector>
+
+#include <cuml/genetic/common.h>
+#include <cuml/genetic/genetic.h>
+#include <cuml/genetic/program.h>
+#include <cuml/cuml.hpp>
```

Comment:
[this](https://github.com/rapidsai/cuml/pull/3565/) PR seems to have eliminated `cuml.hpp` upon merging with latest branch, so might wanna update here

Reply:
Ah I see, I believe this is also the cause of the error you're getting in `ML::deviceAllocator`.